charts/descheduler/templates/deployment.yaml

```diff
@@ -6,7 +6,14 @@ metadata:
   labels:
     {{- include "descheduler.labels" . | nindent 4 }}
 spec:
+  {{- if gt .Values.replicas 1.0}}
+  {{- if and .Values.leaderElection .Values.leaderElection.enabled }}
```

Comment:
it could also be helpful to check that users aren't doing the inverse, ie setting >1 replicas without leaderelection

Reply:
Actually, the following condition already checks it, isn't it?

```
{{- if gt .Values.replicas 1.0}}
{{- if and .Values.leaderElection .Values.leaderElection.enabled }}
```

it means: _setting >1 replicas without leaderelection_

Reply:
Yup, at least if I'm reading it right, currently this code will fail if `replicas > 1` and `leaderElection != nil`, but we want what you just posted

Reply:
We tested the following scenarios:

1. Works as is
```yaml
replicas: 1
leaderElection: {}
```

2. Got: `You must set leaderElection to use more than 1 replica`
```yaml
replicas: 2
leaderElection: {}
```

3. Got: `You must set leaderElection to use more than 1 replica`
```yaml
replicas: 2
leaderElection:
  enabled: false
```

4. Works smoothly

```yaml
replicas: 2
leaderElection:
  enabled: true
``

5. Works. We allowed this case since user might want to scale their deployment later
```
replicas: 1
leaderElection:
  enabled: true
```

Reply:
We've also added a note that we think may be useful to users.

```
{{- if eq .Values.kind "Deployment" }}
{{- if eq .Values.replicas 1.0}}
WARNING: You set replica count as 1 and workload kind as Deployment however leaderElection is not enabled. Consider enabling Leader Election for HA mode.
{{- end}}
{{- end}}
```